python-package/lightgbm/basic.py

```diff
@@ -408,21 +408,27 @@ def _choose_param_value(main_param_name: str, params: Dict[str, Any], default_va
     # avoid side effects on passed-in parameters
     params = deepcopy(params)
 
-    # find a value, and remove other aliases with .pop()
-    # prefer the value of 'main_param_name' if it exists, otherwise search the aliases
-    found_value = None
+    aliases = set(_ConfigAliases.get(main_param_name)) - {main_param_name}
+
+    # if main_param_name was provided, keep that value and remove all aliases
     if main_param_name in params.keys():
-        found_value = params[main_param_name]
+        for param in aliases:
+            params.pop(param, None)
+        return params
 
-    for param in _ConfigAliases.get(main_param_name):
-        val = params.pop(param, None)
-        if found_value is None and val is not None:
-            found_value = val
+    # if main param name was not found, search for an alias
+    for param in aliases:
+        if param in params.keys():
+            params[main_param_name] = params[param]
+            break
 
-    if found_value is not None:
-        params[main_param_name] = found_value
-    else:
-        params[main_param_name] = default_value
+    if main_param_name in params.keys():
+        for param in aliases:
+            params.pop(param, None)
+        return params
```

Comment:
Isn't this the same as lines 414-417?

Reply:
If `main_param_name in params.keys():` is true in this block, it means "a value was provided via an alias". This needs to be here to clear out the other aliases.

That's different from lines 414-417, which is like "a value was provided via the main parameter name".

Reply:
Ah I see. Maybe after not finding the `main_param_name` we could do the following:
```
aliases_in_params = list(params.keys() & aliases)
# if any alias was found, set the value of the first one
if aliases_in_params:
    params[main_param_name] = params[aliases_in_params[0]]
# remove remaining aliases
for aliases_in_params[1:]:
    params.pop(param, None)
```
I'm approving since this is just a suggestion. Thanks for the explanation!